include/overlay025/poketch_system.h

```diff
@@ -13,73 +13,130 @@
+    u8 subState; // each systemState has its own sub-state machine, they all share this variable
+    u8 appState;
+    u8 touchingScreen;
+    u8 playerMoved;
+    u8 appChanging;
     u8 unk_06;
     u8 pedometerUpdated;
-    u32 unk_08;
-    BOOL unk_0C;
-    u32 unk_10;
+    u32 buttonState;
+    BOOL skipApp;
+    u32 appSkipTimer;
     enum PoketchAppID loadedAppID;
     FSOverlayID loadedAppOverlayID;
     UnkStruct_ov25_02254560 * unk_1C;
     UnkStruct_ov25_02254560_1 unk_20;
     void * unk_24;
-    PoketchButtonManager * unk_28;
+    PoketchButtonManager *buttonManager;
     u32 unk_2C;
     u32 unk_30;
     PoketchSystem ** poketchSysPtr;
     SysTask * unk_38;
-    UnkFuncPtr_ov25_02254238 unk_3C;
-    UnkFuncPtr_ov25_02254238_1 unk_40;
-    UnkFuncPtr_ov25_02254274 unk_44;
-    void * unk_48;
+
+    PoketchAppInitFunction currAppInit;
+    PoketchAppShutdownFunction currAppShutdown;
+    PoketchAppSaveFunction currAppSave;
+    void * appSaveData;
+
     BGL *bgl;
-    NNSG2dOamManagerInstance * unk_50;
-    SaveData * saveData;
-    PoketchData * poketchData;
-    FieldSystem * fieldSystem;
+    NNSG2dOamManagerInstance * oamManager;
+
+    SaveData *saveData;
+    PoketchData *poketchData;
+    FieldSystem *fieldSystem;
     enum ButtonDir buttonDir;
 };
 
 void PoketchSystem_Create(FieldSystem *fieldSystem, PoketchSystem **poketchSys, SaveData *saveData, BGL *bgl, NNSG2dOamManagerInstance *oamManager);
-void ov25_02253D5C(PoketchSystem *poketchSys);
-BOOL ov25_02253D70(PoketchSystem *poketchSys);
-void ov25_02253D7C(PoketchSystem *poketchSys, int param1, u32 param2);
-BOOL ov25_02253DD4(PoketchSystem *poketchSys);
+void PoketchSystem_StartShutdown(PoketchSystem *poketchSys);
+BOOL PoketchSystem_IsSystemShutdown(PoketchSystem *poketchSys);
+void PoketchSystem_SendEvent(PoketchSystem *poketchSys, enum PoketchFieldEventID eventID, u32);
+BOOL PoketchSystem_IsTapped(PoketchSystem *poketchSys);
 enum PoketchAppID PoketchSystem_CurrentAppID(PoketchSystem *poketchSys);
-void ov25_02254238(UnkFuncPtr_ov25_02254238 param0, UnkFuncPtr_ov25_02254238_1 param1);
-void ov25_0225424C(PoketchSystem *poketchSys);
-void ov25_02254260(PoketchSystem *poketchSys);
-void ov25_02254274(UnkFuncPtr_ov25_02254274 param0, void * param1);
+void PoketchSystem_SetAppFunctions(PoketchAppInitFunction initFunction, PoketchAppShutdownFunction shutdownFunction);
+void PoketchSystem_NotifyAppLoaded(PoketchSystem *poketchSys);
+void PoketchSystem_NotifyAppUnloaded(PoketchSystem *poketchSys);
+void PoketchSystem_SetSaveFunction(PoketchAppSaveFunction saveFunction, void *saveData);
 UnkStruct_ov25_02254560 * ov25_02254418(void);
-void ov25_02254424(u32 param0);
+void PoketchSystem_PlaySoundEffect(u32 soundID);
 void ov25_02254444(u32 param0, u32 param1);
-BOOL ov25_0225446C(u32 * param0, u32 * param1);
-BOOL ov25_022544BC(u32 * param0, u32 * param1);
+BOOL PoketchSystem_IsHoldingDisplay(u32 *x, u32 *y);
+BOOL PoketchSystem_TappedDisplay(u32 *x, u32 *y);
```

Comment:
[polish] Thinking that these are more accessors than `Is` checks; maybe `PoketchSystem_GetDisplayHeldCoords` and `PoketchSystem_GetDisplayTappedCoords`?